src/components/CameraScannerLayout.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect } from 'react'
+import { Image } from 'react-native'
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withRepeat,
+  withTiming,
+} from 'react-native-reanimated'
+import Box from './Box'
+
+const SCANNER_SIZE = 300
+const SCANNER_LINE_HEIGHT = 43
+const SCAN_DURATION = 2000
+
+export const CameraScannerLayout = () => {
+  const linePosition = useSharedValue<number>(-SCANNER_LINE_HEIGHT)
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [{ translateY: linePosition.value }],
+  }))
+
+  useEffect(() => {
+    linePosition.value = withRepeat(
+      withTiming(SCANNER_SIZE, {
+        duration: SCAN_DURATION,
+      }),
+      -1,
+    )
+  }, [linePosition])
+
+  return (
+    <Box
+      position="absolute"
+      top={0}
+      left={0}
+      right={0}
+      bottom={0}
+      justifyContent="center"
+      alignItems="center"
+      opacity={0.5}
+      backgroundColor="primaryBackground"
+    >
+      <Box
+        position="relative"
+        width={SCANNER_SIZE}
+        height={SCANNER_SIZE}
+        justifyContent="space-between"
+        alignItems="center"
+        overflow="hidden"
+      >
+        {/* top left */}
+        <Box
+          position="absolute"
+          borderColor="hntBlue"
+          width={40}
```

Comment:
Should we also have a constant for 40?

Reply:
resolved add BORDER_SEGMENT_SIZE constant for 40